session.go

```diff
@@ -39,6 +39,14 @@ type Session interface {
 	ClientQueryBuilder
 	EditInteractionResponse(ctx context.Context, interaction *InteractionCreate, message *Message) error
 	SendInteractionResponse(context context.Context, interaction *InteractionCreate, data *InteractionResponse) error
+
+	CreateGlobalCommand(ctx context.Context, applicationID Snowflake, command *ApplicationCommand) error
+	CreateGuildCommand(ctx context.Context, applicationID Snowflake, guildID Snowflake, command *ApplicationCommand) error
+	UpdateCommandGlobal(ctx context.Context, applicationID, commandID Snowflake, command *ApplicationCommand) error
+	UpdateCommandGuild(ctx context.Context, applicationID, guildID, commandID Snowflake, command *ApplicationCommand) error
+	DeleteCommandGlobal(ctx context.Context, applicationID, commandID Snowflake) error
+	DeleteCommandGuild(ctx context.Context, applicationID, guildID, commandID Snowflake) error
```

Comment:
These new rest methods need to follow the resource based design of disgord. I should have commented on this back when EditInteractionResponse and SendInteractionResponse was first added.

Essentially all these should reside within a ApplicationCommandQueryBuilder.
client.ApplicationCommand(aplicationID).Global().Create(...)
client.ApplicationCommand(aplicationID).Global().Update(...)
client.ApplicationCommand(aplicationID).Global().Delete(...)
client.ApplicationCommand(aplicationID).Guild(guildID).Create(...)
client.ApplicationCommand(aplicationID).Guild(guildID).Update(...)
client.ApplicationCommand(aplicationID).Guild(guildID).Delete(...)

the context.Context needs to be defined in a WithContext(ctx) method as the rest of the QueryBuilders as well.

Reply:
See a short example here https://github.com/andersfylling/disgord/blob/d2b7a7d7b2715c49750dc86f4f6bf2cfefe8e154/role.go#L77